src/transforms/parsing-utilities.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExportSpecifier, ExportNamedDeclaration, ExportDefaultDeclaration } from 'estree';
+import { PluginContext } from 'rollup';
+import { ExportNameToClosureMapping, ExportClosureMapping } from '../types';
+
+type DeclarationsWithFunctions = ExportNamedDeclaration | ExportDefaultDeclaration;
+
+export const exportSpecifierName = (exportSpecifier: ExportSpecifier): string =>
+  exportSpecifier.exported.name;
+
+export function functionDeclarationName(
+  context: PluginContext,
+  declaration: DeclarationsWithFunctions,
+): string | null {
+  // For the Declaration passed, there can be a function declaration.
+  if (declaration.declaration && declaration.declaration.type === 'FunctionDeclaration') {
+    const functionDeclaration = declaration.declaration;
+
+    if (functionDeclaration === null) {
+      context.error(
+        `Plugin requires exports to be named, 'export function Foo(){}' not 'export function(){}'`,
+      );
+    }
+    // This function declaration is the export name we need to know.
+    return functionDeclaration.id && functionDeclaration.id.name
+      ? functionDeclaration.id.name
+      : null;
```

Comment:
may be include a warning for `null` case, as it might not minify

Reply:
Addressed with eca1c68.

Reply:
I thought we were going for a warn here instead of an error.